File: feb/8 feb/lc50.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if 0 > n:
            x = 1/x
            n = -n
        def traverse(n):
            if n == 0:
                return 1
            if n == 1:
                return x
            p = traverse(n//2)
            if n % 2 == 0:
                return p * p
            return p * p * x 
        return traverse(n)
    
print(Solution().myPow(2,-2))

File: june/1june/lc50.java
class Solution {
    double pow(double x,int n){
        if(n == 0) return 1.0;
        if(n == 1) return x;
        double d = pow(x,n/2);
        if(n%2 == 0) return d * d;
        return d * d * x;
    }
    public double myPow(double x, int n) {
        if(n == 0) return 1.0;
        if(n == 1) return x;
        if(0>n){
            x = 1/x;
            n = -1 * n;
        }
        return pow(x,n);
    }
}